packages/fetchai/skills/generic_buyer/behaviours.py

```diff
@@ -92,3 +95,64 @@ def teardown(self) -> None:
         :return: None
         """
         pass
+
+
+class GenericTransactionBehaviour(TickerBehaviour):
+    """A behaviour to sequentially submit transactions to the blockchain."""
+
+    def __init__(self, **kwargs):
+        """Initialize the transaction behaviour."""
+        tx_interval = cast(float, kwargs.pop("tx_interval", DEFAULT_TX_INTERVAL))
+        self.max_processing = cast(
+            float, kwargs.pop("max_processing", DEFAULT_MAX_PROCESSING)
+        )
+        self.processing_time = 0.0
+        self.waiting: List[Tuple[LedgerApiDialogue, LedgerApiMessage]] = []
+        self.processing: Optional[LedgerApiDialogue] = None
+        super().__init__(tick_interval=tx_interval, **kwargs)
+
+    def setup(self) -> None:
+        """Setup behaviour."""
+        pass
+
+    def act(self) -> None:
+        """
+        Implement the act.
+
+        :return: None
+        """
+        if self.processing is not None and self.processing_time <= self.max_processing:
+            # already processing
+            self.processing_time += self.tick_interval
+            return
+        if len(self.waiting) == 0:
+            # nothing to process
+            return
+        self.start_processing()
+
+    def start_processing(self) -> None:
+        """Process the next transaction."""
+        dialogue, message = self.waiting.pop(0)
```

Comment:
Minor: I suggest putting this as private method. The reason is, it only makes sense if called from act, because before the `pop` there isn't check of emptiness.
So either you move the check of emptiness inside `start_processing`, or put as private? Another is, put the check in both parts.